indico/modules/events/papers/operations.py

```diff
@@ -189,13 +187,12 @@ def reset_paper_state(paper):
 
 def _store_paper_template_file(template, file):
     content_type = mimetypes.guess_type(file.filename)[0] or file.mimetype or 'application/octet-stream'
-    filename = secure_filename(file.filename, 'template')
```

Comment:
Just to confirm, why was this check removed? Is it safe to trust the file?

Reply:
No, it's a name coming from the user. But we still apply it for the filename that is used for storage on the file system (or whatever storage backend is used). It's just no longer applied to the file name stored in the database, so when you download a file stored in Indico later it will give you the same filename used during upload.

Reply:
All right, sounds good.

Reply:
FWIW it's not unsafe per se, but I did notice that it lets you upload garbage files containing `../` or `/` in general which would then break the download links since the filenames are not supposed to contain either of those. So I'll add a new util that sanitizes a filename for display purposes and strips this stuff while keeping non-ASCII and spaces.

Reply:
Updated to strip this kind of garbage anyway while preserving useful stuff.